sdk/state/payment.go

```diff
@@ -80,19 +80,30 @@ func (c *Channel) ProposePayment(amount int64) (CloseAgreement, error) {
 
 var ErrUnderfunded = fmt.Errorf("account is underfunded to make payment")
 
-// ConfirmPayment confirms a close agreement. The original proposer should only have to call this once, and the
-// receiver should call twice. First to sign the agreement and store signatures, second to just store the new signatures
-// from the other party's confirmation.
-func (c *Channel) ConfirmPayment(ca CloseAgreement) (closeAgreement CloseAgreement, authorized bool, err error) {
-	// If the close agreement details don't match the close agreement in progress, error.
+func (c *Channel) validatePayment(ca CloseAgreement) (err error) {
```

Comment:
I usually prefer breaking out validations, but in the case of the `ConfirmPayment` function validation can't be done entirely prior to signature verification. The balance field is validated during verification. This means this function does not completely validate a payment.

Reply:
I wonder if we can change the way the balance validations work. I think we can if we introduce some way of tracking who sender of a payment is. I'll give this some more thought. This problem is fine to ignore for now I think.

Reply:
ya I noticed that as well that the entire validation wouldn't be in this function. I thought moving the bulk of the validation to a different method was worth the trade off though, since `ConfirmPayment` was getting pretty bloated imo. I can add a comment explaining that, and then we can remove if we find a good way of having all the validations in this method

Reply:
https://github.com/stellar/experimental-payment-channels/pull/183/commits/423700c2602323a4309bd19d3b65e6815fa1eb7d